### PR TITLE
[Event Hubs Client] AmqpConsumer Enhancement

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to apply some of the Service Bus learnings to the `AmqpConsumer` for better efficiency and fewer allocations due to the closure.

# Last Upstream Rebase

Tuesday, March 30, 5pm (EST)

# References and Related

- #19884

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs [hunk 1]
```diff
@@ -236,52 +236,56 @@ internal class AmqpConsumer : TransportConsumer
...
+                            default
                         ).ConfigureAwait(false);
 
                         cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
                         // If event messages were received, then package them for consumption and
                         // return them.
 
-                        if ((messagesReceived) && (amqpMessages != null))
+                        foreach (AmqpMessage message in messagesReceived)
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs:262]: I was just curious about the perf impacts of removing the conditional check before doing the `foreach` and it's quite a bit slower (though very fast in absolute time)

```
|                 Method |     Mean |     Error |    StdDev | Ratio | RatioSD | Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------- |---------:|----------:|----------:|------:|--------:|------:|------:|------:|----------:|
| ConditionalEnumeration | 1.132 ns | 0.0147 ns | 0.0137 ns |  1.00 |    0.00 |     - |     - |     - |         - |
|          DoEnumeration | 5.490 ns | 0.0459 ns | 0.0429 ns |  4.85 |    0.08 |     - |     - |     - |         - |
```

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs:262]: Interesting.

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs:262]: I tweaked to return `null` when nothing was received and added a check to bail out early if we have none.

**Review by christothes — APPROVED** [review]

**Pull request merged**